libbeat/keystore/keystore.go

```diff
@@ -39,39 +39,45 @@ var (
 // to that concept, so we can deal with tokens that has a limited duration or can be revoked by a
 // remote keystore.
 type Keystore interface {
-	// Store add keys to the keystore, wont be persisted until we save.
-	Store(key string, secret []byte) error
-
 	// Retrieve returns a SecureString instance of the searched key or an error.
 	Retrieve(key string) (*SecureString, error)
 
-	// Delete removes a specific key from the keystore.
-	Delete(key string) error
-
-	// List returns the list of keys in the keystore, return an empty list if none is found.
-	List() ([]string, error)
-
 	// GetConfig returns the key value pair in the config format to be merged with other configuration.
 	GetConfig() (*common.Config, error)
 
-	// Create Allow to create an empty keystore.
-	Create(override bool) error
-
 	// IsPersisted check if the current keystore is persisted.
 	IsPersisted() bool
+}
+
+type WritableKeystore interface {
+	Keystore
+	// Store add keys to the keystore, wont be persisted until we save.
+	Store(key string, secret []byte) error
+
+	// Delete removes a specific key from the keystore.
+	Delete(key string) error
+
+	// Create Allow to create an empty keystore.
+	Create(override bool) error
 
 	// Save persist the changes to the keystore.
 	Save() error
 }
 
+type ListingKeystore interface {
+	WritableKeystore
+	// List returns the list of keys in the keystore, return an empty list if none is found.
+	List() ([]string, error)
+}
+
 // Packager defines a keystore that we can read the raw bytes and be packaged in an artifact.
 type Packager interface {
 	Package() ([]byte, error)
 	ConfiguredPath() string
 }
 
 // Factory Create the right keystore with the configured options.
-func Factory(cfg *common.Config, defaultPath string) (Keystore, error) {
+func Factory(cfg *common.Config, defaultPath string) (ListingKeystore, error) {
```

Comment:
I think this Factory method is specific to `file_keystore`, perhaps we should move it to `file_keystore.go`

Reply:
Also I saw `ResolverFromConfig`, I think it's not really used? I would remove it if that's the case